### ISSUE TITLE
Health check timeout when instance is healthy.

### ISSUE THREAD

**Issue by Onlinehead** [metabase/metabase#26266]
**Describe the bug**
 `api/health` endpoint became slow which cause timeout on the health checks.
Sometimes endpoint unable to return the result in 3-5 seconds, but at the same time Metabase successfully process other queries related to a regular load. We run probes once per 15 seconds and found that sometimes Metabase failure at least 3 checks one by one (after that our system mark it unhealthy and restart).
We didn't found any strict dependency between CPU usage and health check failures. Sometimes it stop to response even if CPU if almost free (less than 30% usage), sometimes it works more or less ok (but unstable) even under almost full load. Overall it looks like kind of interlock or so. 
For example, we have a Metabase container which use a significant amount of CPU during fingerprinting. CPU limited to 4 cores of the 8 in total which VM has.
![image](https://user-images.githubusercontent.com/1026551/200312179-fd95c8f7-ea50-4dfb-bbb7-ec67b5e974b9.png)
CPU throttling in that case is below 2 ms in 60 seconds average, usually less than 500 us, so almost no throttling at all.
During that process, the response time floating between 200 ms to 1.5 seconds, measured by `time curl localhost:3000/api/health`, with some spikes up to 2+ seconds:

```
# time curl localhost:3000/api/health
{"status":"ok"}
real	0m0.701s
user	0m0.003s
sys	0m0.003s
# time curl localhost:3000/api/health
{"status":"ok"}
real	0m1.508s
user	0m0.000s
sys	0m0.005s
# time curl localhost:3000/api/health
{"status":"ok"}
real	0m2.289s
user	0m0.004s
sys	0m0.002s
```
It lagging, but the response time still not constantly high which can cause a crash (but anyway it is too high to call it "normal", because background task should not slow down the app so heavy). 

**Logs**
No logs which can be linked to that bug. 

**To Reproduce**
1. Run queries on a metabase instance in parallel. For Jetty pool = 50 about 20-30 in parallel should be enough. Queries should be relatively heavy (at least 2-3 seconds to get the result), but do not overload the instance, it should have a free space in the pool. CPU usage should be **not** a bit more than full usage of the one core (60% for 2 available cores, 30% for 4 available cores etc.) to make sure resource shortage will not effect the situation.
2. Measure the response time for the endoint.

**Expected behavior**
Endpoint should give a response with a correct code in up to 500ms. High load doesn't mean the instance is unhealthy if it has free CPU resources and still processing client requests.

**Comment by dpsutton** [comment]
I've solved "half" of this. I've removed the requirement that the health endpoint needs to get a db connection to report if there is activity on the instance. I'm hoping that because we use an async handler this will remove pressure on this endpoint. If not we can look at the [QosFilter](https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/servlets/QoSFilter.html) to prioritize requests on that endpoint.
